torch/csrc/distributed/c10d/Ops.cpp

```diff
@@ -27,7 +28,41 @@ c10::intrusive_ptr<ProcessGroup::Work> allreduce_(
   return process_group->allreduce(
       tensor_vec,
       AllreduceOptions{
-          static_cast<ReduceOp>(reduce_op),
+          ReduceOp(static_cast<ReduceOp::Kind>(reduce_op)),
+          std::chrono::milliseconds(timeout)});
+}
+
+// NCCL specific PREMUL_SUM Reductions
+// NCCL 2.11.1-1 introduced "User Defined Reduction Operators":
+//   https://docs.nvidia.com/deeplearning/nccl/archives/nccl_21212/user-guide/docs/api/ops.html
+c10::intrusive_ptr<ProcessGroup::Work>
```

Comment:
Are the two new PG methods required or just out of convenience for users?

Introducing those methods in the backend agnostic PG class is not a good idea.

They can be exposed as helpers in distributed_c10.py that do the same.